src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,124 @@
+package duke.ui;
+
+import duke.DukeException;
+import duke.parser.Parser;
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * Represents the client facing UI to handle inputs and displays.
+ */
+public class Ui {
+    /**
+     * Adds task to TaskList and displays added task info to user.
+     * @param tasks TaskList
+     * @param task Task to be added
+     */
+    public void addTask(TaskList tasks, Task task) {
+        print("Got it. I've added this task:");
+        print(String.format("  %s\n", task.toString()));
+        print(String.format("Now you have %d tasks in the list.", tasks.size()));
+    }
+
+    /**
+     * Displays bye message.
+     */
+    public void bye() {
+        print("Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Displays all tasks from TaskList that exist on specified date.
+     * @param tasks TaskList
+     * @param date Date given by user
+     * @throws DukeException if invalid date
+     */
+    public void date(TaskList tasks, String date) throws DukeException {
```

Comment:
Maybe could be more specific in naming, mentioning that the date is already converted? Might potentially cause confusion in the future